File: Adjacent Increasing Subarrays Detection II/Adjacent Increasing Subarrays Detection II.java
// Solution for Adjacent Increasing Subarrays Detection II in JAVA

class Solution {
    public int maxIncreasingSubarrays(List<Integer> nums) {
        int prev = 0, curr = 1, maxK = 0;
        for (int i = 1; i < nums.size(); i++) {
            if (nums.get(i) > nums.get(i - 1)) curr++;
            else { prev = curr; curr = 1; }
            maxK = Math.max(maxK, Math.max(curr / 2, Math.min(prev, curr)));
        }
        return maxK;
    }
}

File: Adjacent Increasing Subarrays Detection II/Adjacent Increasing Subarrays Detection II.py
# Solution for Adjacent Increasing Subarrays Detection II in PY

class Solution:
    def maxIncreasingSubarrays(self, nums: list[int]) -> int:
        prev = 0
        curr = 1
        max_k = 0
        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                curr += 1
            else:
                prev, curr = curr, 1
            max_k = max(max_k, curr // 2, min(prev, curr))
        return max_k
